src/triage/component/catwalk/model_grouping.py

```diff
@@ -0,0 +1,160 @@
+import copy
+import json
+import logging
+
+DEFAULT_KEYS = [
+    'label_timespan',
+    'label_name',
+    'as_of_date_frequency',
+    'max_training_history',
+    'state',
+    'cohort_name',
+    'feature_groups'
+]
+
+
+class ModelGrouper(object):
+    """Assign a model group id to given model input based on default or custom configuration
+
+    This class is a wrapper around the `get_model_group_id` stored procedure,
+    which interfaces with the model_groups table provision a stable model group id .
+    The role of this class is mainly to provide data conversion, sensible defaults, and
+    an abstraction layer over the database.
+
+    Args:
+        model_group_keys (list) A list of matrix metadata keys to uniquely define a model group.'
+            In addition, the non-matrix attributes 'class_path' and 'parameters', referring to
+            classifier training arguments, can be sent.
+    """
+    def __init__(self, model_group_keys=[]):
+        self.model_group_keys = model_group_keys
```

Comment:
I'm just hanging out, not really reviewing; but, note, [Codacy](https://app.codacy.com/app/rayidghani/triage/pullRequest?prid=1332468) did a good job! I guess it didn't want to place a comment; but, it's right, function signature defaults should never be mutable.

Moreover, it looks to me like what you _might_ want here is a `set` – you check whether it contains certain elements, remove elements, _etc._. And, it doesn't _appear_ that element order matters.

For example:

    def __init__(self, model_group_keys=()):
        self.model_group_keys = frozenset(model_group_keys)

But since you want to modify a copy, you could as easily –

    def __init__(self, model_group_keys=()):
        self.model_group_keys = model_group_keys

    …

    keys = set(self.model_group_keys)